pyorbital/tlefile.py

```diff
@@ -294,36 +294,54 @@ def _get_local_tle_path_from_env():
 
 def _get_uris_and_open_func(tle_file=None):
     """Get the uri's and the adequate file open call for the TLE files."""
-    def _open(filename):
-        return io.open(filename, "rb")
-
     local_tle_path = _get_local_tle_path_from_env()
 
     if tle_file:
-        if isinstance(tle_file, io.StringIO):
-            uris = (tle_file,)
-            open_func = _dummy_open_stringio
-        elif "ADMIN_MESSAGE" in tle_file:
-            uris = (io.StringIO(read_tle_from_mmam_xml_file(tle_file)),)
-            open_func = _dummy_open_stringio
-        else:
-            uris = (tle_file,)
-            open_func = _open
+        uris, open_func = _get_tle_file_uris_and_open_method(tle_file)
     elif local_tle_path:
-        # TODO: get the TLE file closest in time to the actual satellite
-        # overpass, NOT the latest!
-        list_of_tle_files = glob.glob(local_tle_path)
+        uris, open_func = _get_local_uris_and_open_method(local_tle_path)
+    else:
+        uris, open_func = _get_internet_uris_and_open_method()
+    return uris, open_func
+
+
+def _get_tle_file_uris_and_open_method(tle_file):
+    if isinstance(tle_file, io.StringIO):
+        uris = (tle_file,)
+        open_func = _dummy_open_stringio
+    elif "ADMIN_MESSAGE" in tle_file:
+        uris = (io.StringIO(read_tle_from_mmam_xml_file(tle_file)),)
+        open_func = _dummy_open_stringio
+    else:
+        uris = (tle_file,)
+        open_func = _open
+    return uris, open_func
+
+
+def _open(filename):
+    return io.open(filename, "rb")
+
+
+def _get_local_uris_and_open_method(local_tle_path):
+    # TODO: get the TLE file closest in time to the actual satellite
+    # overpass, NOT the latest!
+    list_of_tle_files = glob.glob(local_tle_path)
+    if list_of_tle_files:
         uris = (max(list_of_tle_files, key=os.path.getctime), )
         LOGGER.debug("Reading TLE from %s", uris[0])
         open_func = _open
     else:
-        LOGGER.debug("Fetch TLE from the internet.")
-        uris = TLE_URLS
-        open_func = urlopen
+        LOGGER.warning("TLES environment variable points to no TLE files")
```

Comment:
Possibly one improvement: Could we mention that instead we fetch tle from celestrak which is throttled so the user should us this functionality sparingly?

Reply:
Should the warning be in the `_get_internet_uris_and_open_method()` so it will be shown always or in `_get_local_uris_and_open_method()` and warn only when `TLES` is empty?

Reply:
Not sure, maybe the latter?